docs/trace-command.md

```diff
@@ -0,0 +1,117 @@
+# Trace Command
+
+## Overview
+
+Applications deployed with Crossplane usually contains several Crossplane Custom Resources (CRs). Additionally 
+Crossplane applies the Kubernetes pattern for Persistent Volume (PV) claims and classes which results in multiple 
+resources which are closely related. While this enables separation of concern between app teams and cluster administrators, 
+it also makes debugging and troubleshooting uneasy. A problem in the bottom of the resource hierarchy affects upper 
+layers and finding actual root cause requires couple of consecutive commands both to see the status and to get the 
+reference of the next resource. 
+
+For example, when a `KubernetesApplicationResource` reports a failure in its status indicating that there is something 
+wrong with accessing Kubernetes cluster, user needs to first find the corresponding `Resource Claim`, check its status
+and may need to continue with the `Managed Resource`, `Resource Class` and `Provider` resources.
+
+Trace command aims to ease debugging and troubleshooting process by providing a holistic view for the resource being 
+traced. 
+
+## Usage
+
+The trace tool is part of the crossplane kubectl plugin and can be used as follows:
+
+```bash
+$ kubectl crossplane trace --help
```

Comment:
Maybe this is in scope for this since it's a documentation pull request, but @displague reports that the tool displays the following output if no arguments are provided:

```
$ kubectl crossplane trace
missing arguments: SUBCOMMAND TYPE[.GROUP] NAME [-n|--namespace NAMESPACE]
```

Would we be able to use `trace` instead of `SUBCOMMAND`, or just print out the full usage if required arguments are missing?

Reply:
Done